src/chessboard/context/chessboard-context.tsx

```diff
@@ -134,14 +134,20 @@ export const ChessboardProvider = forwardRef(
       isDraggablePiece = () => true,
       getPositionObject = () => {},
       onArrowsChange = () => {},
+      onCheckForPromotion = (sourceSquare, targetSquare, piece, targetPiece) => {
+        const columnDifference = Math.abs(sourceSquare.charCodeAt(0) - targetSquare.charCodeAt(0))
+        return ((piece[0] === "w" && sourceSquare[1] === "7" && targetSquare[1] === "8" && 
+                ((!columnDifference && !targetPiece) || (columnDifference === 1 && targetPiece?.charAt(0) === "b"))) ||
+                (piece[0] === "b" && sourceSquare[1] === "2" && targetSquare[1] === "1" && 
+                ((!columnDifference && !targetPiece) || (columnDifference === 1 && targetPiece?.charAt(0) === "w"))))
+      },
```

Comment:
Removing that will mean there are a few strange outcomes, like dragging a white pawn onto another white piece will still show the promotion dialog. Is that alright?

Reply:
as we cannot know what the user wants and whether that move will actually be valid in a few moves time, I think it's best we allow it and the user can override and disable it if they wish, as it can be perfectly valid if used in tandem with premoves

Reply:
Okay, I agree